### ISSUE TITLE
Transaction SAVEPOINT support

### ISSUE THREAD

**Issue by revmischa** [prisma/prisma#12898]
## Problem

I would like to use the SQL [SAVEPOINT](https://en.wikipedia.org/wiki/Savepoint) (nested transaction) feature for nesting transactions.

My use case: running integration tests inside of isolated transactions. More details in https://github.com/prisma/prisma/issues/9710

I would like to temporarily replace `BEGIN` statements with `SAVEPOINT` statements for code executed in my test. The idea is that I want to wrap each test inside its own transaction so it's isolated and doesn't see data from any other tests running at the same time, and issue a `ROLLBACK` at the end so the test never actually commits anything to the DB.

To make this work I need to disable any code running inside the test from beginning a new transaction, instead creating a savepoint. More details can be seen in #9710.

## Suggested solution

Something like SQLAlchemy perhaps: https://docs.sqlalchemy.org/en/14/orm/session_transaction.html#using-savepoint

## Alternatives

Currently I have to run integration tests one by one and truncate every table at the end of each test. This is horribly slow and inefficient and prevents me from running my tests in parallel. 

## Additional context

https://github.com/prisma/prisma/issues/9710

**Comment by codepunkt** [comment]
@Valerionn It does not work with MySQL as MySQL doesn't allow sending `SAVEPOINT` queries as prepared statements and everything prisma allows is sending exactly those.

**Comment by aseemk** [comment]
(I'm aware there are duplicate issues here, but not sure which one(s) the Prisma team is paying most attention to, so sharing this comment I wrote in another issue too. Apologies if that's not what I should do!)

Strong +1. Just hit & discovered this limitation, and it's a major bummer. Hard to have modular application code (e.g. helper functions that can do things atomically, creating and committing their own transactions to do so) and decouple higher-level transactions from lower-level ones.

Besides the testing use case, another use case for higher-level transactions is for robust idempotency across your whole application. Brandur Leach describes this technique really well in this article:

https://brandur.org/idempotency-keys

And ironically, I was thinking of going this way in part because Prisma also doesn't support `SELECT ... FOR UPDATE`! (https://github.com/prisma/prisma/issues/8580) 😕 

Please consider supporting nested transactions via savepoints. Thank you!
